src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -196,6 +198,18 @@ public ResponseEntity<SnapshotBuilderGetConceptsResponse> getConcepts(
     return ResponseEntity.ok(snapshotBuilderService.getConceptChildren(id, conceptId));
   }
 
+  @Override
+  public ResponseEntity<SnapshotBuilderCountResponse> getSnapshotBuilderCount(
+      UUID id, SnapshotBuilderCountRequest body) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    iamService.verifyAuthorization(
```

Comment:
If you change this to `verifyDatasetAuthorization(userRequest, id.toString(), IamAction.VIEW_SNAPSHOT_BUILDER_SETTINGS);` this will throw a 404 error instead of a 401 when the dataset doesn't exist